src/lib/charts/ChartBase.tsx

```diff
@@ -1,44 +1,57 @@
 import React, { FC, createRef } from 'react';
-import { ChartOptions, ChartTooltipModel, ChartType } from 'chart.js';
 import classNames from 'classnames';
-import ChartComponent from 'react-chartjs-2';
+import { ChartTooltipModel, ChartType } from 'chart.js';
+import ChartComponent, { ChartComponentProps } from 'react-chartjs-2';
 import { renderChartTooltip } from './ChartTooltip/ChartTooltip';
 import { merge } from 'lodash';
 import {
   ChartJSData,
+  ChartJSOptions,
   ChartRefComponent,
   ChartTooltipComponent,
   ChartLegendComponent,
   ChartJSDataFunction
 } from './chart.helpers';
 import { ChartLegend } from './ChartLegend/ChartLegend';
 
-export interface ChartBaseProps {
+export interface ChartBaseProps extends Omit<ChartComponentProps, 'data'> {
   chartJSData: ChartJSData | ChartJSDataFunction;
   className?: string;
-  options?: ChartOptions;
-  tooltip?: ChartTooltipComponent;
-  legend?: ChartLegendComponent;
+  options?: ChartJSOptions;
+  tooltipComponent?: ChartTooltipComponent;
+  legendComponent?: ChartLegendComponent;
```

Comment:
I changed this to `legendComponent` because `ChartComponentProps` already has a `legend` prop, and this was giving me a typescript error because the two types were incompatible.

I also went ahead and change the `tooltip` to `tooltipComponent` for consistency.